R/decideVariableLots.R

```diff
@@ -0,0 +1,237 @@
+  }
+  if (options$lsl) {
+    decision_table$addColumnInfo(name = "col_5", title = "Lower Specification Limit (LSL)", type = "number")
+  }
+  if (options$usl) {
+    decision_table$addColumnInfo(name = "col_6", title = "Upper Specification Limit (USL)", type = "number")
+  }
+  if (options$lsl) {
+    decision_table$addColumnInfo(name = "col_7", title = "Z.LSL", type = "number")
+  }
+  if (options$usl) {
+    decision_table$addColumnInfo(name = "col_8", title = "Z.USL", type = "number")
+  }
+  decision_table$addColumnInfo(name = "col_9", title = "Critical Distance (k)", type = "number")
+
+  lotContainer[["decision_table"]] <- decision_table
+  # Sanity checks for sample statistics
+  if (sd_sample <= 0) {
+    lotContainer$setError(sprintf("Sample standard deviation has to be greater than 0."))
+    return ()
+  }
+  
+  if (is.null(mean_sample)) {
+    lotContainer$setError(sprintf("Sample mean is invalid."))
+    return ()
+  }
+  # We always have a sample standard deviation.
+  # We might or might not have a historical standard deviation, so comparison is to be done accordingly.
+  sd_compare <- sd_sample
+  sd <- "unknown"
+  sd_historical <- 0
+  if (options$sd) {
+    # Historical standard deviation is known. We'll use it for comparison.
+    sd <- "known"
+    sd_historical <- options$stdev
+    sd_compare <- sd_historical
+  }
+
+  z.lsl <- NULL
+  z.usl <- NULL
+  decision <- NULL
+
+  if (!options$lsl && !options$usl) {
+    decision <- NULL
+  }
+
+  # Decision for the lot is made based on the available specification limits.
+  # 1. LSL is available.
+  if (options$lsl) {
+    z.lsl <- ((mean_sample - options$lower_spec) / sd_compare)
+    z.lsl <- round(z.lsl, 3)
+    if (!options$usl) {
+      # Only LSL available. Accept/reject lot based on Z.LSL.
+      decision <- z.lsl >= k
+    }
+  }
+  # 2. USL is available.
+  if (options$usl) {
+    z.usl <- ((options$upper_spec - mean_sample) / sd_compare)
+    z.usl <- round(z.usl, 3)
+    if (!options$lsl) {
+      # Only USL available. Accept/reject lot based on Z.USL.
+      decision <- z.usl >= k
+    }
+  }
+  # 3. Both LSL and USL are available.
+  if (options$lsl && options$usl) {
+    if (options$upper_spec < options$lower_spec) {
+      lotContainer$setError(sprintf("USL can not be lower than LSL."))
+      return ()
+    }
+    # When both LSL and USL are specified, we need to decide based on standard deviation.
+    # Historical standard deviation known
+    if (options$sd) {
+      # Error handling for AQL/RQL
+      aql <- round(options$aql, 3)
+      rql <- round(options$rql, 3)
+      if (aql >= rql) {
+        lotContainer$setError(sprintf("AQL (Acceptable Quality Level) value should be lower than RQL (Rejectable Quality Level) value."))
+        return ()
+      }
+      z.p <- (options$lower_spec - options$upper_spec) / (2 * sd_historical)
+      p <- pnorm(z.p)
```

Comment:
```suggestion
      p <- stats::pnorm(z.p)
```
R is a bit pedantic about these things, but you need to either do `#'@importFrom stats pnorm` to import functions using Roxygen2 or you need to prefix functions with their package name. You also need to do this for packages like `stats` which are typically loaded by default (but not for `base`).

Reply:
Added the import statement for the stats module dependencies!